napari/components/camera.py

```diff
@@ -78,3 +120,21 @@ def calculate_nd_view_direction(
         view_direction_nd = np.zeros(ndim)
         view_direction_nd[list(dims_displayed)] = self.view_direction
         return view_direction_nd
+
+    @no_type_check
+    def __setattr__(self, name, value):
```

Comment:
I'll leave a comment on the main thread that I don't love overwriting `__setattr__` - but if we are going to go with this approach then I'd like us to make something like `SettableEventedModel` that lives alongside our `EventedModel` and gets used here. I feel like this pattern is far to generic to live on the camera and if we like it here we might want to start using it elsewhere. If we really like it, then it could just live on `EventedModel` directly - thoughts @tlambert03?